zebra/rt_netlink.c

```diff
@@ -3744,6 +3744,13 @@ static int netlink_ipneigh_change(struct nlmsghdr *h, int len, ns_id_t ns_id)
 	if (tb[NDA_LLADDR]) {
 		/* copy LLADDR information */
 		l2_len = RTA_PAYLOAD(tb[NDA_LLADDR]);
+		if (l2_len != sizeof(mac)) {
+			if (IS_ZEBRA_DEBUG_KERNEL)
+				zlog_debug(
+					"   Neighbor Entry received has a NDA_LLADDR of length %u but was expecting %lu, ignoring",
+					l2_len, sizeof(mac));
+			return 0;
```

Comment:
We shouldn't return here as it breaks the following code which work when `l2_len == 4`.
I think we should just skip `memcpy` when `l2_len != sizeof(mac)` but don't stop processing.

Reply:
NDA_LLADDR can also be valid for NHRP when size is 4 bytes (remote IP endpoint).
This code will break nhrp behaviour.
that piece of code however can be pushed after calling (zsend_nhrp_neighbor_notify )

Reply:
so the netlink code is fundamentally broken for anything other than a 6 byte mac address.  This does not just include this bit of code.  It's throughout the entire file where we handle NDA_LLADDR as must be 6 bytes.

We should stop processing because we don't know how to handle it.  And as I understand it.  A all 0 mac address will invoke some different processing in evpn.  I would rather not turn a neighbor entry into something it is not.

There is not anything specific in our code that needs a 4 byte reachability address at all yet.  In fact I would prefer we don't even pretend to handle 4 bytes until someone comes in and a) has a use case for needing it and b) actually spends the time to get it working properly in FRR.  Until then handling a 4 byte neighbor address when the entirety of the rest of the code assumes 6 bytes is a recipe for some sort of bug to prop up. 

What we do *need* to do here is stop the crashing.  Hence this code.

Reply:
Here's what I think we should do: https://github.com/FRRouting/frr/pull/9686

Reply:
> There is not anything specific in our code that needs a 4 byte reachability address at all yet. In fact I would prefer we don't even pretend to handle 4 bytes until someone comes in 

NHRP uses it !